File: jade/src/main/java/ua/nure/world/PathNode.java
package ua.nure.world;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class PathNode {

    public PathNode parent;

    public List<PathNode> childs;

    public PathNode(PathNode parent) {
        this.parent = parent;
        childs = new ArrayList<>();
    }

    public void addChild(PathNode... child) {
        childs.addAll(Arrays.asList(child));
    }

}

File: scalaactors/src/main/scala/ua/nure/world/PathNode.scala
package ua.nure.world


class PathNode {

    var parent: PathNode

    var childs: List[PathNode]

    def PathNode(parent: PathNode) {
        this.parent = parent
    }

    def addChild(child: List[PathNode]) {
        childs  = List.concat(childs, child)
    }

}
